packages/api/src/external/hie/set-doc-query-progress-with-source.ts

```diff
@@ -108,18 +73,79 @@ export async function appendDocQueryProgressWithSource({
     return updatedPatient;
   });
 
-  await processDocQueryProgressWebhook({
+  await handleWebhookBeingSent({
     patient: result.dataValues,
-    documentQueryProgress: result.data.documentQueryProgress,
     requestId,
   });
 
   return result;
 }
 
+export function aggregateAndSetHieProgresses(
+  existingPatient: PatientModel,
+  updatedExternalData: PatientExternalData,
+  requestId: string
+) {
+  const documentQueryProgress = !existingPatient.data.documentQueryProgress
+    ? {}
+    : existingPatient.data.documentQueryProgress;
+
+  existingPatient.data.externalData = updatedExternalData;
+
+  // Set the aggregated doc query progress for the patient
+  const externalQueryProgresses = setDocQueryProgressWithExternal(updatedExternalData);
+
+  const aggregatedDocProgress = aggregateDocProgress(externalQueryProgresses);
+
+  const updatedDocumentQueryProgress = {
+    ...documentQueryProgress,
+    ...aggregatedDocProgress,
+  };
+
+  updatedDocumentQueryProgress.requestId = requestId;
+
+  const updatedPatient = {
+    ...existingPatient,
+    data: {
+      ...existingPatient.data,
+      documentQueryProgress: updatedDocumentQueryProgress,
+    },
+  };
+
+  return updatedPatient;
```

Comment:
Also, I think this piece of code should be on the one that works with Sequelize. Reason being, this function doesn't know that we have to create a new `data` property otherwise Sequelize won't send an update to the DB, and this is a good thing, so it can be small and cohesive.

Maybe we can return the consolidated DQ progress and let the code within the DB tx do its thing and create a new `data`?

Reply:
Yea it was mostly to share this same logic between the set and tally doc query functions but this makes sense